main.tf

```diff
@@ -206,13 +235,13 @@ resource "azurerm_log_analytics_workspace" "main" {
 }
 
 resource "azurerm_log_analytics_solution" "main" {
-  count = var.log_analytics_workspace_enabled && var.log_analytics_solution_id == null ? 1 : 0
```

Comment:
This removes the sole usage of this variable, and now `tflint` is reporting it as `terraform_unused_declarations`.

So, I'm going to remove it.

Reply:
I would prefer to preserve `var.log_analytics_solution_id` in this expression because this variable was introduced at first place in case the caller want to use log analytics workspace and solution they brought here instead of creating one. Remove this variable will be considered as a breaking change and we cannot introduce a breaking change in a minor version upgrade, which means this pr can only be merged in v7.0.0.

In this case I would prefer the original expression, which is:

```hcl
count = var.log_analytics_workspace_enabled && var.log_analytics_solution_id == null ? 1 : 0
```